extension/icu/icu-strptime.cpp

```diff
@@ -485,12 +486,14 @@ struct ICUStrftime : public ICUDateFunc {
 	}
 };
 
-void RegisterICUStrptimeFunctions(ClientContext &context) {
+void RegisterICUStrptimeFunctions(DatabaseInstance &instance) {
+	ICUStrftime::AddBinaryTimestampFunction("strftime", instance);
+}
+
+void RegisterICUStrptimeCasts(ClientContext &context) {
 	ICUStrptime::AddBinaryTimestampFunction("strptime", context);
 	ICUStrptime::AddBinaryTimestampFunction("try_strptime", context);
 
-	ICUStrftime::AddBinaryTimestampFunction("strftime", context);
-
 	// Add string casts
 	ICUStrptime::AddCasts(context);
```

Comment:
Why are casts still using the context?

Reply:
In https://github.com/duckdb/duckdb/commit/6351f9498e29dc91f4b4b47b8fdbcc039333f810 I tried replacing `ClientContext` with `DatabaseClient`, but I am not sure what I wrote actually make sense, in particular the Bind function changed and I am not sure if that preserves the same meaning.

If there is a way to correctly write the transformation, they should also be ported over

Reply:
And the other thing using the ClientContext are the iteration on the locales, at the very beginning of LoadInternal, and also that I am unsure if it can be reworked in some way to do without context.